### ISSUE TITLE
Crashing on Exception '[Errno 61] No data available'

### ISSUE THREAD

**Issue by nepto** [rdiff-backup/rdiff-backup#20]
Under certain circumstances, **rdiff-backup** tries to remove an ACLs from the file, while there are already no ACLs.

There is no need to bail on this simple non-fatal condition.

Attached patch made against latest version from Git fixes this. Let me know if you want me to check this into repository.

Thanks,
Ondrej

[rdiff-backup-enodata-fix.patch.txt](https://github.com/sol1/rdiff-backup/files/1243661/rdiff-backup-enodata-fix.patch.txt)

**Comment by nepto** [comment]
Still current.

**Comment by jmtd** [comment]
I've just started to see this in my backup logs for one backup in particular (version 1.2.8, debian package -7), I'll try this patch.

**Comment by anomaly256** [comment]
Hi all, sorry for the delay in addressing this.  @nepto feel free to submit PR's in the future if you have any more contributions, I believe this is the workflow we will move to for rdiff-backup.

@jmtd can you confirm this patch fixed the issue you were having?

Thanks

**Comment by jmtd** [comment]
Hi - I had completely forgotten about this! Indeed @anomaly256, I still have this applied and I have had no further backup errors. The thoroughness-stickler in me thinks I should make sure to test some restores too, which I haven't, so I'm going to try to figure out which of my backups was triggering it, and do some restores.

**Comment by jmtd** [comment]
Restore seems fine.

**Comment by anomaly256** [comment]
Thanks for the confirmation, I'll merge this shortly

**Comment by inducer** [comment]
Any progress on this? I've also seen a similar crash with this traceback:

```
Exception '[Errno 61] No data available' raised of class '<type 'exceptions.IOError'>':                                                                                                                                 
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/robust.py", line 32, in check_common_error                                                                                                                                      
    try: return function(*args)                                                                                                                                                                                                               
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/rpath.py", line 1149, in append                                                                                                                                                         
    return self.__class__(self.conn, self.base, self.index + (ext,))                                                                                                                                                     
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/rpath.py", line 884, in __init__                                                                                                                                                        
    else: self.setdata()                                                                                                                                                                                           
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/rpath.py", line 909, in setdata                                                                                                                               
    if self.lstat(): self.conn.rpath.setdata_local(self)                                                                                                                                                             
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/rpath.py", line 1496, in setdata_local                                                                                                                                 
    if Globals.eas_conn: rpath.data['ea'] = ea_get(rpath)                                                                                                                                                                            
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/eas_acls.py", line 597, in rpath_ea_get                                                                                                                                          
    ea.read_from_rp(rp)                                                                                                                                                                                                               
  File "/usr/lib/python2.7/dist-packages/rdiff_backup/eas_acls.py", line 60, in read_from_rp                                                                                                                       
    attr_list = rp.conn.xattr.listxattr(rp.path, rp.issym())     
IOError: [Errno 61] No data available  
```

Which suggests that more spots in that file need to catch and ignore `ENODATA`. To clarify the impact of this: Because of this bug, I have been unable to back up a critical machine.

**Comment by pjeby** [comment]
@inducer: Adding ENODATA to the list of error values on line 62 of eas_acls.py gets rid of that error for me too.  (in addition to the patch attached to this issue)

**Comment by ottok** [comment]
Filed in Debian as https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=881347 and fixed in upload of 1.3.3.

**Comment by ericzolf** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1494236#c7 tells us the probable reason for the error and gives a slightly different patch. I think, it is more correct to continue on such error than to return: there might be further extended attributes in the list that require to be removed.
It also explains why I didn't get the issue with the packaged rdiff-backup on Fedora but only on my Python3 version.

As a side note, as far as I understand, http://xfs.org/docs/xfsdocs-xml-dev/XFS_Labs/tmp/en-US/html/ch04s02.html tells me that it's a specific XFS issue.

I'll push a fix to my python3 branch to fix this issue, no need to do it on master, it will just create merge conflicts.
